src/app.js

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/env node
+// shebang
+
+// const { createCanvas, loadImage, Image } = require('canvas')
+const canvas = require('canvas');
+
+const fs = require('fs');
+// we need fs to read the image & redraw it onto canvas
+
+const yargs = require('yargs');
+// yargs allows us to configure for arguments 
+// we need to accept multiple arguments for business info (5x mandatory & 3x optional) and a JSON object (mandatory)  
+// we need to accept 1 x mandatory argument for JSON file path 
+// we need to accept 1 x optional argument for Locale tag (language) 
+
+const options = yargs
+.usage("Usage: -n <name>")
+.option("o", {alias: "order", describe: "JSON file path", type: "string", demandOption: true})
+.option("n", {alias: "name", describe: "Your name - 20 char limit", type: "string", demandOption: true})
+.option("a", {alias: "address1", describe: "Number, Street, City - 53 char limit", type: "string", demandOption: true})
+.option("b", {alias: "address2", describe: "Country & UK postcode - 53 char limit", type: "string", demandOption: true})
+.option("v", {alias: "vat", describe: "VAT number", type: "string", demandOption: true})
+.option("i", {alias: "image", describe: "url to image for logo", type: "string", demandOption: true})
+.option("t", {alias: "telephone", describe: "Telephone number", type: "number", demandOption: false })
+.option("w", {alias: "website", describe: "Website - 48 char limit", type: "string", demandOption: false})
+.option("t", {alias: "topMessage", describe: "Top Message - 30 char limit", type: "string", demandOption: false})
+.option("l", {alias: "locale", describe: "locale tag", type: "string", demandOption: false})
+.choices("l", ["en", "de", "zh"])
+.argv;
+ 
+// TOO: The Order info: Class that takes JSON file path - do we need to use fs function fs.createReadFileStream
+
+// check
+console.log(`Business name: ${options.name}, Address: ${options.address1 + " "  + options.address2}, VAT: ${options.vat}, Logo/Image URL: ${options.image}`);
+
+// validates mandatory CLI arguments with char limit
```

Comment:
Validation should only be for states that are unrecoverable. I think it's feasible here to truncate the string to the maximum char limit, and log a warning that this has been done, as it's a bit more user friendly. Although I can see the rationale for both ways! yargs actually does a lot of the validation for you so perhaps this is less work than I originally thought.

Reply:
Do you mean I should replace the validate function for all CLI arguments with just manipulating the information given (truncating)? 

`const validate = (arg, limit = 53) => {
  try {
    if (options[arg].length > limit) {
      throw new Error( `${arg} exceeds ${limit} character limit. Valid input required.` );
      // throw new Error exposes an error event with two params name & message. It also terminates further execution. 
    }
  } catch (err) {
      console.log(err.name, err.message);
  }
}`